homeassistant/components/songpal/media_player.py

```diff
@@ -271,6 +309,11 @@ async def async_update(self) -> None:
 
             _LOGGER.debug("Active source: %s", self._active_source)
 
+            (
+                self._active_sound_mode,
+                self._sound_modes,
+            ) = await self._get_sound_modes_info()
```

Comment:
Could this be an issue as the list of sound modes is only initialized during the first update / when connection breaks?

You mention in the docstring below that the sound mode might not be available on all inputs/outputs, so I'm wondering what happens when we populate this only during `update()` which happens only during the initial update or if the connection breaks?